yt/utilities/cython_fortran_utils.pyx

```diff
@@ -29,6 +30,10 @@ cdef class FortranFile:
         self.cfile = fopen(fname.encode('utf-8'), 'rb')
         self._closed = False
 
+        if self.cfile is NULL:
+            self._closed = True
+            PyErr_SetFromErrnoWithFilename(FileNotFoundError, fname.encode('utf-8'))
```

Comment:
I can see it works as expected, but this is the first time I review code using Python's C-API directly. Can you explain why you're using `PyErr_SetFromErrnoWithFilename` here and not `PyExc_FileNotFoundError` ?

Reply:
Actually I can just replace this with a regular Python exception raising (see latest commit)